### ISSUE TITLE
timers: every silently ignores errors

### ISSUE THREAD

**Issue by rationalrevolt** [slavapestov/factor#1508]
```factor
1 >string .
```
The above code throws as error at runtime `Generic word length does not define a method for the fixnum class. Dispatching on object: 1`.

The same code passed into a timer as shown below fails silently:
```factor
[ 1 >string . ] 1 seconds every
```

**Comment by mrjbq7** [comment]
What behavior would you like it to have?

**Comment by rationalrevolt** [comment]
I'm thinking we could disable the timer and log an error (not sure where to)?

**Comment by mrjbq7** [comment]
It would be easy to log an error to the console. With the UI open we probably want this and a couple other errors to popup an error window maybe, but that could be a little more involved. Baby steps.  

I'm not sure an error should stop the timer, but maybe fail fast is better than forgiving. Dunno. Maybe there is an internal API for timers that gives you a choice of behavior. Just thinking out loud.

**Comment by bpollack** [comment]
This is actually less about timers and more about all threads. E.g., `[ + ] "foo" spawn` dies instantly, but generates absolutely no output anywhere. That's also what .NET 1.x did, and is widely regarded as a mistake. .NET 2.0 and later changes that rule to be a hard shutdown, which I think is better behavior, and then adds new functionality so that parent thread can subscribe to, and handle, issues in a child thread. This is similar to our `concurrency.messaging.spawn-linked`.

So that makes me think we might want consider something like this:

  * The existing `spawn` word is moved into a private or low-level vocabulary, in semantics if not literally
  * A new `spawn` that behaves like `spawn-linked-to`, with a standard dynamically scoped mailbox (similar to how we handle `input-stream` and `output-stream`), replaces it
  * Users can subscribe to that mailbox if they want to monitor thread deaths. We likely want a default handler that in prod logs and quits, and in debug mode merely logs, but that could be decided separately.

Thoughts?

**Comment by jonenst** [comment]
On linux it does logs to stderr:
Data stack underflow

(U) Quotation: [ set-namestack init-catchstack self quot>> call => stop ]
(O) Word: +
(U) Quotation: [ signal-handler => ]
    Word: signal-handler
(U) Quotation: [
        OBJ-CURRENT-THREAD special-object error-thread set-global
        current-continuation => error-continuation set-global
        [ original-error set-global ] [ rethrow ] bi
    ]

but more safety is good
And yes, in development it should never quit because a thread launched by the user crashed..

**Comment by bpollack** [comment]
Oh, interesting! So part of the issue here is that, on Mac app bundles, we should be binding stderr to write to the Console. I'll open that as a separate bug. But then I'm wondering why your timer wasn't logging.

**Comment by jonenst** [comment]
does
```
 "stderr from factor\n" error-stream get-global [ stream-write ] [ stream-flush ] bi
```
work on a Mac app bundle ?

**Comment by bpollack** [comment]
Nope. Which makes sense; Factor would have to use `NSLog()` to get the desired behavior. So separate bug from yours. I still think the `spawn-linked` concept is the right way to go there though.

**Comment by erg** [comment]
I think ``threaded-server`` should be using ``spawn-linked`` or ``spawn-linked-to`` instead of ``spawn`` to get errors in a mailbox that would report to the ui or at least fill up with errors somehow. I tried to do this yesterday but it didn't work out yet because the code is pretty confusing to me (and I wrote some of it).

**Comment by bjourne** [comment]
Hm.. It probably does work, but the output in the UI listener isn't the same as the real stdout and stderr. In it `output-stream get` and `output-stream get-global` are different objects. @jonenst probably sees the output because he runs Factor in text mode. So if you run `output-stream get output-stream set-global`, errors in threads should show up.
